packages/salesforcedx-vscode-core/src/commands/util/timestampConflictChecker.ts

```diff
@@ -27,11 +27,18 @@ export class TimestampConflictChecker implements PostconditionChecker<string> {
   private isManifest: boolean;
   private messages: ConflictDetectionMessages;
   private isPushOperation: boolean;
+  private isPullOperation: boolean;
```

Comment:
Having these two booleans implies an additional state is being handled in the class.
The caller has full control of third state by passing false for both in the c'tor or letting them default.

What is the third state and is it mutually exclusive of push or pull?

I there a way to express the needed states in a simpler way, 'push' | 'pull' | 'thrird' | ...